src/sif/autocost.cc

```diff
@@ -268,27 +268,29 @@ class AutoCost : public DynamicCost {
   }
 
   /**
-   * Returns a function/functor to be used in location searching which will
+   * Function to be used in location searching which will
    * exclude and allow ranking results from the search by looking at each
    * edges attribution and suitability for use as a location by the travel
-   * mode used by the costing method. Function/functor is also used to filter
+   * mode used by the costing method. It's also used to filter
    * edges not usable / inaccessible by automobile.
    */
-  virtual const EdgeFilter GetEdgeFilter() const {
-    // Throw back a lambda that checks the access for this type of costing
+  float Filter(const baldr::DirectedEdge* edge,
+               const baldr::GraphId& edgeid,
+               const baldr::GraphTile* tile) const override {
+    // take into account live traffic constraints
+    if (IsClosedDueToTraffic(edgeid, tile))
```

Comment:
since this is unliklely and its a heavier check, we should move it into the `if` below and put it as the last conditions. this allows other cheaper checks to fail first and avoid ever going to look at the traffic data. we should do this for all the costings not just this one

Reply:
nice catch